port_forwarding.go

```diff
@@ -0,0 +1,386 @@
+package testcontainers
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types/container"
+	"github.com/google/uuid"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/testcontainers/testcontainers-go/internal/core/network"
+)
+
+const (
+	image string = "testcontainers/sshd:1.1.0"
+	// hostInternal is the internal hostname used to reach the host from the container,
+	// using the SSHD container as a bridge.
+	hostInternal string = "host.testcontainers.internal"
+	user         string = "root"
+	sshPort             = "22"
+)
+
+// sshPassword is a random password generated for the SSHD container.
+var sshPassword = uuid.NewString()
+
+// exposeHostPorts performs all the necessary steps to expose the host ports to the container, leveraging
+// the SSHD container to create the tunnel, and the container lifecycle hooks to manage the tunnel lifecycle.
+// At least one port must be provided to expose.
+// The steps are:
+// 1. Create a new SSHD container.
+// 2. Expose the host ports to the container after the container is ready.
+// 3. Close the SSH sessions before killing the container.
+func exposeHostPorts(ctx context.Context, req *ContainerRequest, p ...int) (ContainerLifecycleHooks, error) {
+	var sshdConnectHook ContainerLifecycleHooks
+
+	if len(p) == 0 {
+		return sshdConnectHook, fmt.Errorf("no ports to expose")
+	}
+
+	// Use the first network of the container to connect to the SSHD container.
+	var sshdFirstNetwork string
+	if len(req.Networks) > 0 {
+		sshdFirstNetwork = req.Networks[0]
+	} else if sshdFirstNetwork == "bridge" && len(req.Networks) > 1 {
```

Comment:
It seems like this branch will never be reached because  `sshdFirstNetwork` is set to the zero value and `len(req.Networks)` overlaps with the first branch. Did you mean to put this into a separate if *after* the first?

Reply:
Indeed. I just realised the second branch came from a previous state. Changing it locally, thanks!